src/api/telemetry/TelemetryAPI.js

```diff
@@ -325,19 +331,40 @@ define([
         this.standardizeRequestOptions(arguments[1]);
         const provider = this.findRequestProvider.apply(this, arguments);
         if (!provider) {
-            return Promise.reject('No provider found');
-        }
+            this.noRequestProviderForAllObjects = this.requestProviders.every(requestProvider => {
```

Comment:
Logic looks good here, thanks. Let's put this and the error message handling into a separate function though, `request()` is getting a little long.